core/common/src/main/java/alluxio/security/authentication/plain/SaslParticipiantProviderPlain.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.security.authentication.plain;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.PropertyKey;
+import alluxio.exception.status.UnauthenticatedException;
+import alluxio.security.LoginUser;
+import alluxio.security.User;
+import alluxio.security.authentication.AuthType;
+import alluxio.security.authentication.AuthenticationProvider;
+import alluxio.security.authentication.SaslParticipiantProvider;
+import alluxio.security.authentication.TransportProviderUtils;
+
+import javax.security.auth.Subject;
+import javax.security.sasl.Sasl;
+import javax.security.sasl.SaslClient;
+import javax.security.sasl.SaslException;
+import javax.security.sasl.SaslServer;
+import java.security.Security;
+import java.util.HashMap;
+import java.util.Set;
+
+/**
+ * Implementation of {@link SaslParticipiantProvider} for plain authentication.
+ */
+public class SaslParticipiantProviderPlain implements SaslParticipiantProvider {
+  static {
+    // Register Sasl server implementation for plain mechanism.
+    Security.addProvider(new PlainSaslServerProvider());
+  }
+
+  @Override
+  public SaslClient createSaslClient(Subject subject) throws UnauthenticatedException {
+    String username = null;
+    String password = "noPassword";
```

Comment:
what is this?

Reply:
I moved it as it is. I suppose, schemes that use Subject have no use of the password.